src/handlers/expenditures/expenditureMadeStaged.ts

```diff
@@ -1,22 +1,21 @@
 import { mutate } from '~amplifyClient';
+import { ExtensionEventListener } from '~eventListeners';
 import {
   ExpenditureType,
   UpdateExpenditureDocument,
   UpdateExpenditureMutation,
   UpdateExpenditureMutationVariables,
 } from '~graphql';
-import { ContractEvent } from '~types';
-import { getExpenditureDatabaseId, output, toNumber, verbose } from '~utils';
+import { EventHandler } from '~types';
+import { getExpenditureDatabaseId, toNumber, verbose } from '~utils';
 
-export default async (event: ContractEvent): Promise<void> => {
-  const { colonyAddress } = event;
+export const handleExpenditureMadeStaged: EventHandler = async (
```

Comment:
not sure how much overhead this would add, but should we type this specific function differently, so it receives an `ExtensionEventListener` in the parameter type definition?

Reply:
Maybe we could have different event handlers typed and reuse those,
so we have a `ExtensionEventHandler` which gets a listener that's of the type `ExtensionEventListener`

Then we can have a general `EventHandler` type which would just be a union of all different types?

Reply:
I've tried that initially and battled it tonight but generics in union types defeated me again. 

The problem here is that on one hand, we want to make listeners discriminated by handler type, but each handler can be used by multiple listener types - e.g. `ExpenditureCancelled` can be emitted by either colony or extension.

The most correct approach would be to narrow down the listener type but it feels like a bit of an overkill (you'd need a function, if check, etc.)